init/main.go

```diff
@@ -697,6 +705,57 @@ func checkIfInitrd() error {
 	return nil
 }
 
+func blkInfoMatchesArg(info *blkInfo, arg string) bool {
+	if strings.HasPrefix(arg, "UUID=") && info.uuid == strings.TrimPrefix(arg, "UUID=") {
+		return true
+	}
+	if strings.HasPrefix(arg, "LABEL=") && info.label == strings.TrimPrefix(arg, "LABEL=") {
+		return true
+	}
+
+	return false
+}
+
+func resume(devpath string) error {
+	resume, ok := cmdline["resume"]
+	if !ok {
+		return nil
+	}
+
+	var resumeDevice string
+	if resume == devpath {
+		resumeDevice = devpath
+		goto resume
+	}
+
+	{
+		info, err := readBlkInfo(devpath)
+		if err != nil {
+			return fmt.Errorf("%s: %v", devpath, err)
+		}
+
+		if blkInfoMatchesArg(info, devpath) {
+			resumeDevice = devpath
+			goto resume
+		}
+	}
+
+	return nil
+resume:
```

Comment:
i dont love this but if there's a larger refactor in the cards I'll address whether or not a goto makes sense here later